### ISSUE TITLE
Impossible to type number in scientific notation (e.g. '2e3') when using FormlyJsonschema

### ISSUE THREAD

**Issue by jdreSAG** [ngx-formly/ngx-formly#3757]
**Description**
I'm creating field config with FormlyJsonschema toFieldConfig method. Form consists of one field- of type number, no validation or required flag. I run the app, my goal is to put '2e3' into input. I try to type in input '2e' and input automatically became blank right after 'e' key is pressed. 
Only way to make it work is to paste '2e3' or to type '23' and then put 'e' between digits to make it '2e2'.

**Minimal Reproduction**
open https://stackblitz.com/edit/angular-a2uvhd?file=src%2Fapp%2Fapp.component.ts and try to type '2e3' in input

**Your Environment**

- Angular version: 14.0.0
- Formly version: 6.0.0

**Additional context**
Issue is not occurring when fields property is declared manually (without FormlyJsonschema toFieldConfig)

**Comment by bozhidardedov** [comment]
Looks like the validator expression added for `number` fields doesn't work when you enter numbers in scientific notation symbol by symbol - https://github.com/ngx-formly/ngx-formly/blob/main/src/core/json-schema/src/formly-json-schema.service.ts#L177-L179
As soon as you enter the `e` sign the expression
```js
case 'number': {
  return typeof value === 'number';
}
```
throws an error
![image](https://github.com/ngx-formly/ngx-formly/assets/95280800/baaab85b-dc77-4d71-9f04-8007bb81c86c)
This explains also why it works when you enter `e` between numbers :)

**Comment by PaulG96** [comment]
This issue also occurs when entering minus signs before a number has been entered i.e. '-'. On Firefox this issue also occurs for decimal points e.g. '1.'.

**Comment by Sterling-Dev** [comment]
I hit a similar issue while upgrading from 5.12.17 to 6.1.8 with Angular 14.2.

I found the control returned null whenever the input wasn't a proper "number", but the built-in field parser would change null to undefined with the `isEmpty(v)` check.
 https://github.com/ngx-formly/ngx-formly/blob/0b006245b4b5517d9407d6295dd8b637cfef52d0/src/core/json-schema/src/formly-json-schema.service.ts#L194C18-L194C18

Then the FormlyField would set the control to undefined from null because `null !== undefined`
https://github.com/ngx-formly/ngx-formly/blob/v6.0.0/src/core/src/lib/components/formly.field.ts#L310

The 5.x version didn't change null to undefined
https://github.com/ngx-formly/ngx-formly/blob/v5.12.7/src/core/json-schema/src/formly-json-schema.service.ts#L158

It is not an elegant solution, but my workaround was to modify the parsers during field generation to add the null check back in to prevent the control value from changing.
```typescript
formlyJsonSchema.toFieldConfig(schema,
{
  map: (f, _) => {
    // Sanity check that only the expected parser is there.
    if ((f.type === 'number' || f.type === 'integer') && f.parsers && f.parsers.length === 1) {
      const baseParser = f.parsers[0];
      f.parsers[0] = (v: string | number | null) => v === null ? v : baseParser(v);
    }
    return f;
  },
});
```

**Comment by aitboudad** [comment]
This issue has been fixed and released as part of [v6.2.0](https://github.com/ngx-formly/ngx-formly/releases/tag/v6.2.0) release.

Please let us know, in case you are still encountering a similar issue/problem.
Thank you!
